components/camel-webhook/src/main/java/org/apache/camel/component/webhook/WebhookEndpoint.java

```diff
@@ -91,11 +98,17 @@ protected void doStart() throws Exception {
     @Override
     protected void doStop() throws Exception {
         super.doStop();
-
         if (configuration.isWebhookAutoRegister() && delegateEndpoint != null) {
             LOG.info("Unregistering webhook for endpoint: {}", delegateEndpoint);
             delegateEndpoint.unregisterWebhook();
         }
+        ServiceHelper.stopService(delegateEndpoint);
```

Comment:
Not sure it is safe to stop the endpoint here.

Reply:
The webhook is creating the delegate endpoint and should handle its lifecycle. That is how its done in other componens that does something like this